strax/processing/general.py

```diff
@@ -364,3 +366,14 @@ def _touching_windows(thing_start, thing_end,
         result[i] = left_i, right_i
 
     return result
+
+def _check_sorted(things, containers):
+    """
+    Check whether things and containers are sorted
+    :param things: Sorted array of interval-like data
+    :param containers: Sorted array of interval-like data
+    """
+    for array, name in zip([things, containers], ['things', 'containers']):
+        assert np.all(np.diff(array['time']) >= 0), f'{name} should be sorted'
+        assert np.all(array['endtime'] - array['time'] >= 0), f'{name} should'\
```

Comment:
This does not check if things are sorted does it? It only checks if an interval is not negative. Things being sorted would be 
```python
np.diff(array['time']) >= 0
```
Some algorithms also require the intervals to be non-overlapping. In that case you could use:
```python 
    assert np.all(array['endtime'][:-1] - array['time'][1:] < 0)
```

Reply:
I intended to make another check that the length of `things` and `containers` are non-negative here.

Reply:
`touching_windows` also needs the `endtime` to be sorted. `split_by_containment` by definition is stricter and requires the intervals to be non-overlapping, because it returns only one container for each thing.

Reply:
I must apologize, I just noticed that you are in fact computing the `diff` here... Somehow I have overlooked this yesterday o.O 

For split by containment we could add the second check I mentioned to test that intervals are indeed non-overlapping.